src/tests/ftest/aggregation/aggregation_basic.py

```diff
@@ -0,0 +1,106 @@
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+import os
+import time
+from ior_test_base import IorTestBase
+from ior_utils import IorCommand, IorMetrics
+
+
+class DaosAggregationBasic(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description:
+
+       Run IOR with same file option to verify the  
+       aggregation happens in the background. Check for the 
+       space usage before and after the aggregation to verify
+       the overwritten space is reclaimed.
+
+    :avocado: recursive
+    """
+
+    def test_basic_aggregation(self):
+        """Jira ID: DAOS-3451
+
+        Test Description:
+            Purpose of this test is to run ior using DFS api to write to 
+            the DAOS and rerun it again with the same file option and 
+            enable the aggregation and verify the aggregation reclaims
+            the overwritten space.
+
+        Use case:
+            Run ior with read, write, CheckWrite, CheckRead
+            Run ior with read, write, CheckWrite, CheckRead with same file
+            Enable aggregation and wait for 90 seconds to let the files
+            old enough to be marked for aggregation.
+            Wait for aggregation to complete.
+            Check the utilized capacity of the container/pool. It should be 
+            roughly the same as after the initial run since aggregation has 
+            reclaimed the overwritten capacity.
+
+        :avocado: tags=all,pr,hw,large,daosio,aggregatebasic
+        """
+
+        self.update_ior_cmd_with_pool()
+        storage_index = 1
+
+        initial_pool_info = self.pool.pool.pool_query()
+        initial_space_usage = initial_pool_info.pi_space.ps_space.s_free[storage_index] 
+
+        #Disable the aggregation
+        self.pool.set_property("reclaim", "disabled")
+
+        # Run ior with -k option to retain the file and not delete it 
+        self.run_ior_with_pool()
+        first_pool_info = self.pool.pool.pool_query()
+        space_use_after_first_ior = first_pool_info.pi_space.ps_space.s_free[storage_index] 
+        self.log.info("Space used after first ior = {}".format(space_use_after_first_ior))
+
+        space_used_by_ior = initial_space_usage - space_use_after_first_ior
+ 
+        # Run ior the second time on the same pool and container, so another
+        # copy of the file is created in DAOS.
+        self.run_ior_with_pool(update=False)
+        second_pool_info = self.pool.pool.pool_query()
+        space_use_after_second_ior = second_pool_info.pi_space.ps_space.s_free[storage_index]
+        self.log.info("Space used after second ior = {}".format(space_use_after_second_ior))
```

Comment:
(pylint-line-too-long) Line too long (92/80)
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments